packages/web-styles/src/placeholders/_color.scss

```diff
@@ -3,20 +3,49 @@
 @use '../variables/color';
 @use '../functions/color' as color-fn;
 
-// CWF color variables not rendered by bootstrap
-%color-variables {
-  --text-color: #{color.$black};
-  --text-color-rgb: #{color-fn.rgb-values(color.$black)};
-  --background-color: #{color.$white};
-  --background-color-rgb: #{color-fn.rgb-values(color.$white)};
+// background rgb properties
+%color-background-rgb {
+  background-color: rgb(var(--bg-rgb)) !important;
+  color: var(--contrast-color) !important;
 }
 
-// Inverted CWF variables for dark backgrounds
-%inverted-colors {
-  --text-color: #{color.$white};
-  --text-color-rgb: #{color-fn.rgb-values(color.$white)};
-  --background-color: #{color.$black};
-  --background-color-rgb: #{color-fn.rgb-values(color.$black)};
+// background rgba properties
+%color-background-rgba {
+  --bg-opacity: 1;
+  background-color: rgba(var(--bg-rgb), var(--bg-opacity)) !important;
+  color: var(--contrast-color) !important;
+}
+
+// background extensions for containers with dynamic content (unlike buttons, badges, etc.)
+%color-background-container-extensions {
```

Comment:
These styles should be default. It's impractical and complicated to decide where to apply this and where it's breaking things. The high specificity of the selector this is creating, is overriding things that should'nt be overriden.

Reply:
I agree with you.
But if we do this, it will result in a CSS overhead for components such as buttons (which we will refactor maybe too with this placeholders in the future).

The Selectors for this are allready hugh and will become even larger when we add this placeholder to more components.
And they are absolutly obsolete for buttons, etc.
```css
.bg-yellow a:not(.btn, .badge), .bg-light a:not(.btn, .badge), .bg-gray a:not(.btn, .badge), .bg-dark a:not(.btn, .badge), .bg-primary a:not(.btn, .badge), .bg-white a:not(.btn, .badge), .bg-success a:not(.btn, .badge), .bg-info a:not(.btn, .badge), .bg-warning a:not(.btn, .badge), .bg-danger a:not(.btn, .badge), .bg-nightblue a:not(.btn, .badge), .bg-nightblue-bright a:not(.btn, .badge), .bg-petrol a:not(.btn, .badge), .bg-petrol-bright a:not(.btn, .badge), .bg-coral a:not(.btn, .badge), .bg-coral-bright a:not(.btn, .badge), .bg-olive a:not(.btn, .badge), .bg-olive-bright a:not(.btn, .badge), .bg-purple a:not(.btn, .badge), .bg-purple-bright a:not(.btn, .badge), .bg-aubergine a:not(.btn, .badge), .bg-aubergine-bright a:not(.btn, .badge), .toast-primary a:not(.btn, .badge), .alert-primary a:not(.btn, .badge), .toast-notification a:not(.btn, .badge), .alert-notification a:not(.btn, .badge), .toast-success a:not(.btn, .badge), .alert-success a:not(.btn, .badge), .toast-danger a:not(.btn, .badge), .alert-danger a:not(.btn, .badge), .toast-error a:not(.btn, .badge), .alert-error a:not(.btn, .badge), .toast-warning a:not(.btn, .badge), .alert-warning a:not(.btn, .badge), .toast-info a:not(.btn, .badge), .alert-info a:not(.btn, .badge), .bg-yellow label, .bg-light label, .bg-gray label, .bg-dark label, .bg-primary label, .bg-white label, .bg-success label, .bg-info label, .bg-warning label, .bg-danger label, .bg-nightblue label, .bg-nightblue-bright label, .bg-petrol label, .bg-petrol-bright label, .bg-coral label, .bg-coral-bright label, .bg-olive label, .bg-olive-bright label, .bg-purple label, .bg-purple-bright label, .bg-aubergine label, .bg-aubergine-bright label, .toast-primary label, .alert-primary label, .toast-notification label, .alert-notification label, .toast-success label, .alert-success label, .toast-danger label, .alert-danger label, .toast-error label, .alert-error label, .toast-warning label, .alert-warning label, .toast-info label, .alert-info label {
  color: inherit;
}

.bg-yellow a:not(.btn, .badge):hover, .bg-light a:not(.btn, .badge):hover, .bg-gray a:not(.btn, .badge):hover, .bg-dark a:not(.btn, .badge):hover, .bg-primary a:not(.btn, .badge):hover, .bg-white a:not(.btn, .badge):hover, .bg-success a:not(.btn, .badge):hover, .bg-info a:not(.btn, .badge):hover, .bg-warning a:not(.btn, .badge):hover, .bg-danger a:not(.btn, .badge):hover, .bg-nightblue a:not(.btn, .badge):hover, .bg-nightblue-bright a:not(.btn, .badge):hover, .bg-petrol a:not(.btn, .badge):hover, .bg-petrol-bright a:not(.btn, .badge):hover, .bg-coral a:not(.btn, .badge):hover, .bg-coral-bright a:not(.btn, .badge):hover, .bg-olive a:not(.btn, .badge):hover, .bg-olive-bright a:not(.btn, .badge):hover, .bg-purple a:not(.btn, .badge):hover, .bg-purple-bright a:not(.btn, .badge):hover, .bg-aubergine a:not(.btn, .badge):hover, .bg-aubergine-bright a:not(.btn, .badge):hover, .toast-primary a:not(.btn, .badge):hover, .alert-primary a:not(.btn, .badge):hover, .toast-notification a:not(.btn, .badge):hover, .alert-notification a:not(.btn, .badge):hover, .toast-success a:not(.btn, .badge):hover, .alert-success a:not(.btn, .badge):hover, .toast-danger a:not(.btn, .badge):hover, .alert-danger a:not(.btn, .badge):hover, .toast-error a:not(.btn, .badge):hover, .alert-error a:not(.btn, .badge):hover, .toast-warning a:not(.btn, .badge):hover, .alert-warning a:not(.btn, .badge):hover, .toast-info a:not(.btn, .badge):hover, .alert-info a:not(.btn, .badge):hover {
  color: rgba(var(--contrast-color-rgb), 0.8);
}
```

What if we create the placeholders like so:
```scss
// bg-colored elements
%color-background-elementonly-rgb {
  background-color: rgb(var(--bg-rgb)) !important;
  color: var(--contrast-color) !important;
}

%color-background-elementonly-rgba {
  --bg-opacity: 1;
  background-color: rgba(var(--bg-rgb), var(--bg-opacity)) !important;
  color: var(--contrast-color) !important;
}

// bg-colored container extensions
%color-background-container-extensions {
  a:not(.btn, .badge),
  label {
    color: inherit;
  }

  a:not(.btn, .badge):hover {
    color: rgba(var(--contrast-color-rgb), 0.8);
  }
}

// bg-colored containers
%color-background-container-rgb {
  @extend %color-background-elementonly-rgb;
  @extend %color-background-container-extensions;
}

%color-background-container-rgba {
  @extend %color-background-elementonly-rgba;
  @extend %color-background-container-extensions;
}
```

Reply:
To generate less css, we decided to remove the placeholder %color-background-container-extensions completly and instead, we set the link and label colors with --contrast-color by default.